docker/run_cypress_tests/Dockerfile

```
@@ -11,13 +11,37 @@ FROM ruby:3.1.4
 ENV RAILS_ENV=production
 EXPOSE 3000
 
-# https://github.com/nodesource/distributions#installation-instructions
-RUN curl -sL https://deb.nodesource.com/setup_14.x | bash - && \
-    curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
-    curl -sS https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add - && echo "deb http://apt.postgresql.org/pub/repos/apt/ `lsb_release -cs`-pgdg main" | tee  /etc/apt/sources.list.d/pgdg.list && \
+# use "--login" option, so that .bashrc is sourced in new shells
+SHELL ["/bin/bash", "--login", "-c"] 
+
+# Install Node.js
+# https://github.com/nodesource/distributions/issues/1583#issuecomment-1597489401
+# https://stackoverflow.com/a/57546198/
+# Unfortunately, we have to explicitly specify the node version here
+# and cannot use 16.x as we need to put the node binary into the PATH
+# and therefore require the exact version to find the folder
+ENV NODE_VERSION=16.20.1
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash
+ENV NVM_DIR=/root/.nvm
+RUN source ${NVM_DIR}/nvm.sh && nvm install ${NODE_VERSION} && \
+    nvm use ${NODE_VERSION} && nvm alias default ${NODE_VERSION}
```

I think it makes sense to quote these to prevent argument splitting, i.e..:
```suggestion
RUN source "${NVM_DIR}/nvm.sh" && nvm install "${NODE_VERSION}" && \
    nvm use "${NODE_VERSION}" && nvm alias default "${NODE_VERSION}"
```